google/cloud/storage/object_metadata.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_OBJECT_METADATA_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_OBJECT_METADATA_H_
+
+#include "google/cloud/storage/internal/common_metadata.h"
+#include <map>
+
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+/**
+ * Represents a Google Cloud Storage Bucket Metadata object.
+ *
+ * @warning This is an incomplete implementation to validate the design. It does
+ * not support changes to the metadata. It also lacks support for ACLs,
+ * encryption keys, and other features.
+ *
+ * TODO(#537) - complete the implementation.
+ */
+class ObjectMetadata : private internal::CommonMetadata {
+ public:
+  ObjectMetadata() = default;
+
+  static ObjectMetadata ParseFromJson(std::string const& payload);
```

Comment:
Objects don't have a `labels` field in their metadata.

This might be helpful; I took the descriptions of object [1] and bucket [2] metadata, then ran it through Python to parse that JSON and grab the top-level keys for each type.  I've included those lists of keys in a comment in #537.

[1] https://cloud.google.com/storage/docs/json_api/v1/objects#resource
[2] https://cloud.google.com/storage/docs/json_api/v1/buckets#resource

Reply:
Ooops, renamed everything except the functions names.  Fixed.  Also created a new issue to track the TODO.

Reply:
Thanks for that delta, super useful!